Add figures to illustrate the bounds decoding correction (#1169)
riscv/riscv-cheri: src/cheri/rvy64-encoding.adoc

`R` marks the lower boundary of the <<section_cap_representable_check,representable range>> (see <<cap_bounds_map>>).
If `A`, `T` or `B` is less than `R`, then it lies in the 2^E+MW^ aligned region above `R`, and the corrections _c~t~_ and _c~b~_ adjust the upper bits of the decoded bounds accordingly.

As shown in xref:section_cap_bounds_format[xrefstyle=short] below, the upper bits of the capability address are used to form the upper bits of the bounds.

> Sometimes the bounds need to be adjusted up or down relative to the capability address, because the memory bound and the capability address cross alignment boundaries (address 0x20000 in <<capformat_correction_illustration>>)

Reviewer: ```suggestion
Sometimes the bounds need to be adjusted up or down relative to the capability address, because the memory bound and the capability address cross alignment boundaries (address 0x20000 in <<capformat_correction_illustration>>).
```

> Sometimes the bounds need to be adjusted up or down relative to the capability address, because the memory bound and the capability address cross alignment boundaries (address 0x20000 in <<capformat_correction_illustration>>).

Reviewer 2: ```suggestion
As shown in xref:section_cap_bounds_format[xrefstyle=short] below, the upper bits of the bound are computed by substituting B or T into the upper bits of the capability address.                          
This substitution is only correct if the memory bound and the capability address lie within the same 2^E+MW^-aligned region.                                                                               
When they do not (address 0x20000 in <<capformat_correction_illustration>>), the correction factors c~t~ and c~b~ adjust the computed upper bits to select the correct region. 
```

How about this wording to avoid using "sometimes" (which I don't think we should have in a specification) and to also avoid "bounds being adjusted". This could be interpreted as adjusting bounds of existing capabilities (e.g. YBNDSW), this makes it more clear that we are changing the way they are being computed.

[#capformat_correction_illustration]

> .Example capability bounds that use the correction. The base address bound is an s-aligned region below the address, meaning the upper bits (above the mantissa) of the base are smaller than the upper bits of the address. Therefore c~b~=-1. The address and top address bound are in the same region, so c~t~=0.

Reviewer 2: I don't think we define `s` anywhere?

Reviewer 2: Oh it's in the image, never mind.

> .Example capability bounds that use the correction. The base address bound is an s-aligned region below the address, meaning the upper bits (above the mantissa) of the base are smaller than the upper bits of the address. Therefore c~b~=-1. The address and top address bound are in the same region, so c~t~=0.

Reviewer 2: ```suggestion
.Example capability bounds that use the correction. The base is an s-aligned region below the address, meaning the upper bits (above the mantissa) of the base are smaller than the upper bits of the address. Therefore c~b~=-1. The address and top address bound are in the same region, so c~t~=0.
```

Author: Ah, I think we used that terminology more heavily in the past, I guess to distinguish from the encoded base and top. I'll remove the references.
